### DISCUSSION TITLE
Carbon has changed default branch name from MASTER to MAIN -- what do I do?

### DISCUSSION THREAD

**Discussion by dakahn** [carbon-design-system/carbon#8016] [Help] [answered]
Carbon has updated our naming for our default branch from Master to Main. This change effects your work with Carbon in your local clones (on your machines) and on your fork of Carbon on Github.com.

## Updating your local clone:
1. `git branch -m master main` <-- renames your local `master` branch to `main`
2. `git fetch origin` <-- get all the latest branches from your remotes
3. `git branch -u origin/main main` <-- switch your local branch to track the new remote
4. `git push origin main` <-- to push your new local `main` branch up to Github

_Currently there is no command line way (to my knowledge, let me know in the comments if this isn't the case) to set a new `default` repo branch on Github.com. So we have to do that manually._

## Updating the default branch of your fork of Carbon on Github:
1. On Github.com select the branches view
![image](https://user-images.githubusercontent.com/40970507/110356833-43284500-8000-11eb-9074-bbff3bac2a1b.png)

2. Click to change the default branch
![image](https://user-images.githubusercontent.com/40970507/110356865-4ae7e980-8000-11eb-9f44-c1f3050d3328.png)

3. In the change default branch menu select the new `main` branch and follow the prompts
![image](https://user-images.githubusercontent.com/40970507/110356901-55a27e80-8000-11eb-92ed-1bb96d05dc45.png)

**Comment by tay1orjones** [comment, marked as answer]
Viewing my fork for the first time since the change, I got this nice prompt to change the default.

![Screen Shot 2021-03-08 at 3 46 51 PM](https://user-images.githubusercontent.com/3360588/110386199-9e206300-8025-11eb-8232-b3db3336c703.png)

Worth noting that the `gh` cli tool now has direct api access and it appears you can change your default branch through it! https://github.com/cli/cli/issues/929#issuecomment-629253585

**Reply by dakahn** [reply]
Oh great!
